File: C/lru_test.c
// Test file includes a few basic tests for the LRU cache
// This file uses the library of functions for using an LRU cache

#include<stdio.h> 
#include "lru.c" 
#include "linkedList.h"

void printCacheCharacteristics(LruCache* cache);

int main() {
    int size = 5;
    LruCache* cache = CreateLRU(size);
    for(int i=0;i<size*2-3;i++){
        putElement(cache,i);
        printCacheCharacteristics(cache);
    }
    printf("\nInserting done\n\n");
    // // for(int i=0;i<size*2;i++){
        printf("%d ",getElement(cache,4));
        printCacheCharacteristics(cache);
        putElement(cache,13);
        printCacheCharacteristics(cache);
        
        printf("%d ",getElement(cache,3));
        printCacheCharacteristics(cache);
    //     printf("%d ",getElement(cache,8));
    //     printCacheCharacteristics(cache);
    //     printf("%d ",getElement(cache,7));
    //     printCacheCharacteristics(cache);
    //     printf("%d ",getElement(cache,8));
    //     printCacheCharacteristics(cache);
    //     printf("%d ",getElement(cache,9));
    //     printCacheCharacteristics(cache);
    //     printf("%d ",getElement(cache,8));
    //     printCacheCharacteristics(cache);
    // // }
    printf("Complete!\n");
    return 0;
}

void printCacheCharacteristics(LruCache* cache){
    printf("\n\nCache:\n\t");
    printDLL(cache->head);
    if(cache->start!=NULL&&cache->end!=NULL){
        printf("\n\nStart: %d\nEnd: %d\nFull? : %d\n",cache->start->key,cache->end->key,cache->full);
    }else{
        printf("\n\nStart: %d\nEnd: %d\nFull? : %d\n",-1,-1,cache->full);
    }
    printf("\n\n-------------------------------------\n\n");
}

File: Go/lru_test.go
package main

import (
	"fmt"
	"testing"
)

func TestLRUCache(t *testing.T) {
	size := 2
	cache := CreateLRU(size)

	PutElement(cache, 1)
	PrintCache(cache)
	PutElement(cache, 2)
	PrintCache(cache)
	GetElement(cache, 1)
	PrintCache(cache)

	PutElement(cache, 3)
	PrintCache(cache)
	GetElement(cache, 2)
	PrintCache(cache)

	PutElement(cache, 4)
	PrintCache(cache)
	GetElement(cache, 1)
	PrintCache(cache)

	GetElement(cache, 3)
	PrintCache(cache)
	GetElement(cache, 4)
	PrintCache(cache)

	fmt.Println("Complete!")
}
